src/lib/tuning/index/base_index_evaluator.cpp

```diff
@@ -0,0 +1,219 @@
+      dynamic_cast<const GDFSCache<std::string, std::shared_ptr<AbstractLQPNode>>*>(&(lqp_cache.cache()));
+  if (!gdfs_cache_ptr) {
+    LOG_WARN("BaseIndexEvaluator can only analyze GDFSCache instances! Evaluations may be useless...");
+    return;
+  }
+
+  const auto& fibonacci_heap = gdfs_cache_ptr->queue();
+
+  LOG_DEBUG("Query plan cache (size: " << fibonacci_heap.size() << "):");
+  if (fibonacci_heap.size() == 0) {
+    LOG_WARN("There are no logical query plans in the cache. Make sure that logical query plans get cached!");
+  }
+
+  auto cache_iterator = fibonacci_heap.ordered_begin();
+  auto cache_end = fibonacci_heap.ordered_end();
+
+  for (; cache_iterator != cache_end; ++cache_iterator) {
+    const auto& entry = *cache_iterator;
+    LOG_DEBUG("  -> Query '" << entry.key << "' frequency: " << entry.frequency << " priority: " << entry.priority);
+    _inspect_lqp_operator(entry.value, entry.frequency);
+  }
+}
+
+void BaseIndexEvaluator::_inspect_lqp_operator(const std::shared_ptr<const AbstractLQPNode>& op,
+                                               size_t query_frequency) {
+  std::list<const std::shared_ptr<const AbstractLQPNode>> queue;
+  queue.push_back(op);
+  while (!queue.empty()) {
+    auto lqp_node = queue.front();
+    queue.pop_front();
+
+    if (auto left_child = lqp_node->left_child()) {
+      queue.push_back(left_child);
+    }
+    if (auto right_child = lqp_node->right_child()) {
+      queue.push_back(right_child);
+    }
+
+    switch (lqp_node->type()) {
+      case LQPNodeType::Predicate: {
+        auto predicate_node = std::dynamic_pointer_cast<const PredicateNode>(lqp_node);
+        DebugAssert(predicate_node, "LQP node is not actually a PredicateNode");
+        auto lqp_ref = predicate_node->column_reference();
+        if (lqp_ref.original_node()) {
+          auto original_node = lqp_ref.original_node();
+          auto original_columnID = original_node->find_output_column_id(lqp_ref);
+          if (original_node->type() == LQPNodeType::StoredTable) {
+            DebugAssert(original_columnID, "Could not find column ID for LQPColumnReference");
+            auto stored_table = std::dynamic_pointer_cast<const StoredTableNode>(original_node);
+            DebugAssert(stored_table, "referenced node is not actually a StoredTableNode");
+
+            AccessRecord access_record(stored_table->table_name(), *original_columnID, query_frequency);
+            access_record.condition = predicate_node->predicate_condition();
+            access_record.compare_value = boost::get<AllTypeVariant>(predicate_node->value());
+            _access_records.push_back(access_record);
+          }
+        }
+      } break;
+      case LQPNodeType::Join:
+        // Probably interesting, room for future work
+        break;
+      default:
+        // Not interesting
+        break;
+    }
+  }
+}
+
+void BaseIndexEvaluator::_inspect_pqp_operator(const std::shared_ptr<const AbstractOperator>& op,
```

Comment:
Is this ever used? Could you explain the reasoning behind that?

Reply:
This is not used anymore as we now can obtain LQPs from the cache. This is still here in order to offer a fallback, should it be decided that LQPs are not cached anymore. However, as it is not used right now, we could just delete that method. Opinions?